src/librustdoc/html/render/print_item.rs

```diff
@@ -958,6 +958,21 @@ fn item_trait(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item, t: &clean:
     let cloned_shared = Rc::clone(&cx.shared);
     let cache = &cloned_shared.cache;
     let mut extern_crates = FxHashSet::default();
+
+    if !t.is_object_safe(cx.tcx()) {
+        write_small_section_header(
+            w,
+            "object-safety",
+            "Object Safety",
+            &format!(
+                "<div class=\"object-safety-info\">This trait is <b>not</b> \
+                <a href=\"{base}/reference/items/traits.html#object-safety\">\
+                object safe</a>.</div>",
+                base = crate::clean::utils::DOC_RUST_LANG_ORG_CHANNEL
+            ),
```

Comment:
For the anchor:

```suggestion
            &format!(
                "<h3 id=\"{id}\">Object safety<a href=\"#{id}\" class=\"anchor\">§</a></h3>\
                 This trait is <b>not</b> \
                 <a href=\"{base}/reference/items/traits.html#object-safety\">\
                 object safe</a>.",
                base = crate::clean::utils::DOC_RUST_LANG_ORG_CHANNEL,
                id = "object-safety",
             ),
```

Then in the test, just check for the ID presence. One thing you'll need to update too will be to add `object-safety` ID into `init_id_map` (in `html/markdown.rs`).

Reply:
`write_small_section_header` already does all of this, I just added a test for the ID